db/warehouse/migrate/20241207185501_create_ce_referrals.rb

```diff
@@ -0,0 +1,88 @@
+    end
+
+    # TBD we probably want some kind of association between pools an requirements
+    # create_table(:ce_match_pool_requirements) do |t|
+    #   t.references :candidate_pool, null: false, foreign_key: { to_table: :ce_match_policies }, index: false
+    #   t.references :requirement, null: false, foreign_key: { to_table: :ce_match_requirements}, index: false
+    #   t.timestamps
+    # end
+
+    create_table(:ce_match_candidates) do |t|
+      t.references :candidate_pool, null: false, foreign_key: { to_table: :ce_match_candidate_pools }, index: false
+      t.references :client, null: false, foreign_key: { to_table: :Client }
+      t.integer :priority_score, null: true
+      t.timestamps
+      t.index [:candidate_pool_id, :client_id], unique: true, name: 'index_ce_match_candidates_uniq'
+    end
+
+    # Opportunities (vacancies or services within projects)
+    create_table(:ce_opportunities) do |t|
+      t.references :candidate_pool, null: true, foreign_key: { to_table: :ce_match_candidate_pools }
+      t.references :project, null: false # what project provides this opportunity?
+      t.string :workflow_template_identifier, null: false # use an identifier to allow
+      t.string :name, null: false
+      t.string :status, null: false
+      t.datetime :expires_at
+      t.timestamps
+    end
+
+    create_table(:ce_opportunity_categories) do |t|
+      t.string :name, null: false
+      t.timestamps
+    end
+
+    create_table(:ce_opportunity_categorizations) do |t|
+      t.references :opportunity, null: false, foreign_key: { to_table: :ce_opportunities }, index: false
+      t.references :category, null: false, foreign_key: { to_table: :ce_opportunity_categories }
+      t.index [:opportunity_id, :category_id], unique: true, name: 'index_ce_opportunity_categorizations_uniq'
+      t.timestamps
+    end
+
+    # Referrals (instances of clients going through workflow)
+    create_table(:ce_referrals) do |t|
+      t.references :opportunity, null: false, foreign_key: { to_table: :ce_opportunities }
+      t.references :workflow_instance, null: false, foreign_key: { to_table: :wfe_instances }
+      t.string :status, null: false
+      t.references :client, null: false
+      t.references :referred_by
+      t.datetime :completed_at
+      t.timestamps
+    end
+
+    create_table(:ce_referral_participants) do |t|
+      t.references :referral, null: false, foreign_key: { to_table: :ce_referrals }
+      t.references :user, null: false
+      t.references :swimlane, null: true, foreign_key: { to_table: :wfd_swimlanes }
+      t.timestamps
+    end
+
+    create_table(:ce_referral_notes) do |t|
+      t.references :referral, null: false, foreign_key: { to_table: :ce_referrals }
+      t.references :participant, null: false, foreign_key: { to_table: :ce_referral_participants }
+      t.jsonb :submitted_form_data
+      t.timestamps
+    end
```

Comment:
did we decide we needed these at the referral level or only the step level? (I thought step.)

Reply:
Confirmed that notes will be tied to the referral for MVP, not tied to an individual step

Reply:
I do wonder about the complexity of tying the note to a `ce_referral_participants` rather than directly to `users`. 2 thoughts on that:
* There may be some administrative user who isn't a part of any swimlane, but is still able to go in and leave a note (?)
* The user isn't necessarily making that note "in the capacity of" a swimlane, I don't think? Asked another way, if the user is a member of 2 swimlanes (assuming that's possible) how do you decide which ce_referral_participants to link to their note?